pkg/logql/log/pattern/pattern.go

```diff
@@ -6,40 +6,67 @@ import (
 )
 
 var (
-	ErrNoCapture   = errors.New("at least one capture is required")
-	ErrInvalidExpr = errors.New("invalid expression")
+	ErrNoCapture         = errors.New("at least one capture is required")
+	ErrCaptureNotAllowed = errors.New("named captures are not allowed")
```

Comment:
This is debatable, but I think it makes sense: if user tries to parse _fields_, it's better to return an explicit error. In the future, the restriction can be removed.

Note that we're using an unnamed placeholder <_>. I think we could use a new capture identifier (such as <*>) to emphasize the difference with the pattern parse stage. However, it feels like a new syntax for pattern matching, which may confuse users. I would like to hear others' thoughts on this.

Reply:
I don't have strong opinitions using _ vs * but I definitively agree we should fail fast.

Reply:
I would vote for `<_>` as the behavior between these two examples feels consistent to me:

```
|> "dur=<_> err=<_>"
| pattern "dur=<_> err=<_>"
```

I think of that as telling Loki to ignore any of the content in `<_>`

I'm not sure I see the reason to use `<*>` or maybe I'm misunderstanding something?

Reply:
This popped up in the Slack conversation, so I brought it here.

Thank you, guys, for sharing your thoughts. I keep `<_>`.